contrib/database/sql/conn.go

```diff
@@ -167,8 +179,28 @@ func WithSpanTags(ctx context.Context, tags map[string]string) context.Context {
 	return context.WithValue(ctx, spanTagsKey, tags)
 }
 
+// injectComments returns the query with sql comments injected according to the comment injection mode along
+// with a span id injected into sql comments. The returned span id should be used when the sql span is created
+// following the traced database call.
+func injectComments(ctx context.Context, query string, mode tracer.SQLCommentInjectionMode) (cquery string, spanID uint64) {
+	// The sql span only gets created after the call to the database because we need to be able to skip spans
+	// when a driver returns driver.ErrSkip. In order to work with those constraints, a new span id is generated and
+	// used during SQL comment injection and returned for the sql span to be used later when/if the span
+	// gets created.
+	var spanCtx ddtrace.SpanContext
+	if span, ok := tracer.SpanFromContext(ctx); ok {
+		spanCtx = span.Context()
```

Comment:
I may be slightly confused here, but if there is a span in this context, which has a SpanID attached, and then that ID is returned for `tryTrace` to use it when creating the next span, wouldn't that result in a duplicate SpanID? I think we need a test for that case, unless I'm missing something.